src/zenml/steps/entrypoint_function_utils.py

```diff
@@ -235,17 +236,40 @@ def _validate_input_value(
             parameter: The function parameter for which the value was provided.
             value: The input value.
         """
-        config_dict = ConfigDict(arbitrary_types_allowed=False)
-
-        # Create a pydantic model with just a single required field with the
-        # type annotation of the parameter to verify the input type including
-        # pydantics type coercion
-        validation_model_class = create_model(
-            "input_validation_model",
-            __config__=config_dict,
-            value=(parameter.annotation, ...),
-        )
-        validation_model_class(value=value)
+        annotation = parameter.annotation
+
+        # Handle Optional types
+        origin = get_origin(annotation)
+        if origin is Union:
+            args = get_args(annotation)
+            if type(None) in args:
+                if value is None:
+                    return  # None is valid for Optional types
+                # Remove NoneType from args as this case is handled from here
+                args = tuple(arg for arg in args if arg is not type(None))
+                annotation = args[0] if len(args) == 1 else Union[args]
+
+        # Handle None values for non-Optional types
+        if value is None and annotation is not type(None):
+            raise ValueError(f"Expected {annotation}, but got None")
+
+        # Use Pydantic for all types to take advantage of its coercion abilities
+        try:
+            config_dict = {"arbitrary_types_allowed": True}
+            validation_model_class = create_model(
+                "input_validation_model",
+                __config__=type("Config", (), config_dict),
+                value=(annotation, ...),
+            )
+            validation_model_class(value=value)
+        except ValidationError as e:
+            raise ValueError(f"Invalid input: {e}")
```

Comment:
Why? This validation error was caught in other places to raise a much better error message then you're giving here